sycl/source/backend.cpp

```diff
@@ -38,8 +38,7 @@ static const PluginPtr &getPlugin(backend Backend) {
   case backend::ext_oneapi_cuda:
     return pi::getPlugin<backend::ext_oneapi_cuda>();
   default:
-    throw sycl::runtime_error{"getPlugin: Unsupported backend",
-                              PI_ERROR_INVALID_OPERATION};
+    throw sycl::exception(sycl::errc::runtime, "getPlugin: Unsupported backend");
```

Comment:
I wonder what constructor of `sycl::exception` it uses for the current version? Either way, I would feel better if we explicitly convert it to a `std::error_code` using `sycl::make_error_code`:

```suggestion
    throw sycl::exception(sycl::make_error_code(sycl::errc::runtime), "getPlugin: Unsupported backend");
```